src/sqlfluff/core/templaters/dbt.py

```diff
@@ -259,13 +259,16 @@ def process(self, *, fname, in_str=None, config=None, formatter=None):
         except DbtCompilationException as e:
             # Increment the counter
             self._sequential_fails += 1
-            return None, [
-                SQLTemplaterError(
-                    f"dbt compilation error on file '{e.node.original_file_path}', {e.msg}",
-                    # It's fatal if we're over the limit
-                    fatal=self._sequential_fails > self.sequential_fail_limit,
-                )
-            ]
+            if e.node:
```

Comment:
This change fixes an unrelated bug I discovered: For some errors, a `DbtCompilationException` object has a `node` value of `None`, in which case referencing `e.node.original_file_path` fails. Worse, this prevents the user from seeing the original error message.